linalg/solvers.hpp

```diff
@@ -65,24 +65,95 @@ class IterativeSolverMonitor
 /// Abstract base class for iterative solver
 class IterativeSolver : public Solver
 {
+public:
+   /** @brief Settings for the output behavior of the iterative solver. Guarantees to
+       supress all outputs by default.
+     */
+   struct PrintOptions
+   {
+      /** A summary of the solver process will be reported after the last
+          iteration
+        */
+      bool summary = false;
+      /** If a fatal problem has been detected some context-specific
+          information will be reported
+        */
+      bool errors = false;
+      /** If a non-fatal problem has been detected some context-specific
+          information will be reported
+        */
+      bool warnings = false;
+      /// Detailed information about each iteration will be reported
+      bool iteration_details = false;
+   };
+
+
 #ifdef MFEM_USE_MPI
 private:
    int dot_prod_type; // 0 - local, 1 - global over 'comm'
-   MPI_Comm comm;
+   MPI_Comm comm = MPI_COMM_NULL;
 #endif
 
 protected:
    const Operator *oper;
-   Solver *prec;
+   Solver *prec; //< Left preconditioner
    IterativeSolverMonitor *monitor = nullptr;
 
-   int max_iter, print_level;
-   double rel_tol, abs_tol;
+   /** @name Report
+      These options control the internal reporting behavior into ::mfem::out
+      and ::mfem::err of the iterative solvers.
+     */
+   ///@{
+
+   /** @brief Legacy print level definition, which is left for compatibility with
+       custom iterative solvers.
+
+       See #print_options for more information.
+     */
+   int print_level;
```

Comment:
Should probably explicitly mark this as deprecated.

Reply:
I see the point but not sure how to accomplish this one properly. I just put the MFEM_DEPRECATED back before it, but I still have to touch this variable for backwards compatibility reasons - which makes the CI fail.

Reply:
Probably for now we should just deprecate the setter?

Reply:
I am fine with this change (see my comment there). I will merge this later and try to come up with a more portable suppression of the deprecation message in the setters.